src/audio/eq_fir/eq_fir.c

```diff
@@ -42,21 +42,12 @@
 #endif
 
 /* tracing */
-#define trace_eq(__e, ...) trace_event(TRACE_CLASS_EQ_FIR, __e, ##__VA_ARGS__)
-#define trace_eq_with_ids(comp_ptr, __e, ...)			\
-	trace_event_comp(TRACE_CLASS_EQ_FIR, comp_ptr,		\
-			 __e, ##__VA_ARGS__)
-
-#define tracev_eq(__e, ...) tracev_event(TRACE_CLASS_EQ_FIR, __e, ##__VA_ARGS__)
-#define tracev_eq_with_ids(comp_ptr, __e, ...)			\
-	tracev_event_comp(TRACE_CLASS_EQ_FIR, comp_ptr,		\
-			  __e, ##__VA_ARGS__)
-
-#define trace_eq_error(__e, ...) \
-	trace_error(TRACE_CLASS_EQ_FIR, __e, ##__VA_ARGS__)
-#define trace_eq_error_with_ids(comp_ptr, __e, ...)		\
-	trace_error_comp(TRACE_CLASS_EQ_FIR, comp_ptr,		\
-			 __e, ##__VA_ARGS__)
+
+/* redefine trace class to use common comp_ trace macros */
+#ifdef TRACE_CLASS_COMP
+#undef TRACE_CLASS_COMP
+#endif
+#define TRACE_CLASS_COMP TRACE_CLASS_EQ_FIR
```

Comment:
If we really want to have that loose dependency, it'd be better to not have `#undef` and let it fail on warn, cos it means that something is wrong if you have to redefine it. (But then we will have problem with comps that define traces in headers...)

Personally I don't like that loose dependency at all. Macro shouldn't depend on some symbol that comes from `somewhere`. But I guess you also don't want to redefine the same macros that differ just by class name over and over to mark that it has dependency on some symbol that contains chosen class name. We can use macro here like `make_traces(some_prefix, TRACE_CLASS_*)` that generates said set of macros, but metaprogramming in C is also *meh*.

I like idea of these traces at all, but it's not mergable with how `TRACE_CLASS_COMP` works atm, even as temporary solution.

Reply:
Of course the "make_traces" like idea isn't unheard of in this project (see the imx interrupt driver which has some reasonably similar).

So we need to figure out how these traces ought to work. I wonder if certain inline functions can be moved out of headers so the traces (except generic component traces) aren't even used in headers. Any performance risks in that @jajanusz ?

Reply:
@jajanusz TRACE_CLASS_COMP is already defined at that place, so how the warning would help? I explained why I would not maintain any replacement that does not introduce another overhead better than that since we should reduce this to a common TRACE_CLASS_COMP on that level soon.

Reply:
Having specific classes means that you can enable and disable components at will. So the question is if we want to be able to tune this enabling/disabling per component or it's fine to do per class.

Reply:
@paulstelian97 Xtensa arch doesn't like calls + replacing small functions (like 1 op inline) with call will increase size few times. In ideal world we wouldn't have these inlines in headers, but cos of perf/size we cannot remove them on bigger scale.

Reply:
@mmaka1 Maybe it's better to just remove TRACE_CLASS. It requires too much maintenance (in fw/logger/testbench) and it's not useful at all (we have well structured components already and you can easily tell from where log is coming).

Reply:
@jajanusz Ah fair, some inlines will thus remain there for the forseeable future. However, traces themselves aren't extremely lightweight (I think the trace function itself isn't inline?) so at least moving some traces out of there might be warranted? Except for error case that is, and that sends us back to the issue of deciding which component is active...

Reply:
> @mmaka1 Maybe it's better to just remove TRACE_CLASS. It requires too much maintenance (in fw/logger/testbench) and it's not useful at all (we have well structured components already and you can easily tell from where log is coming).

Removing trace classes outright will be interesting in terms of userspace compatibility as well as the ability to selectively enable and disable traces.

Reply:
@paulstelian97 I believe we want to implement filtering per class. However using statically defined class id per each component will not last long because of reasons I tried to explain in the initial description. A better temporary (1st step) solution might be just to use TRACE_CLASS_COMP by all components and remove those weak redefines. A logger side filtering still could be done by grepping location name while future dsp side fine filtering for individual component implementations should be based on uuid retrieved from the dev object.

Reply:
Hmmm, good point. Anyway this separation is reasonable to do for the component ones but what will we do with componentless traces that simply have to remain componentless (there is no reasonable way to attach a component)?

Reply:
@paulstelian97 Imo, any code that is part of the component (dai) implementation is either executed in context of the initialized component instance or its driver instance. Any other static initializers will probably disappear once the system moves towards dynamic component/dai loading/discovery. And uuid is a property of the driver, so new `..._cl_....()` would require the driver ptr and could retrieve the uuid.

Reply:
> @paulstelian97 Imo, any code that is part of the component (dai) implementation is either executed in context of the initialized component instance or its driver instance. Any other static initializers will probably disappear once the system moves towards dynamic component/dai loading/discovery. And uuid is a property of the driver, so new `..._cl_....()` would require the driver ptr and could retrieve the uuid.

Hm, alright, this sounds good to me. Driver instances, that sounds interesting (you'd group off statics into a structure or something, that would also have the classes?).